### DISCUSSION TITLE
Debug automations

### DISCUSSION THREAD

**Discussion by webserviceXXL** [Budibase/budibase#5193] [Help] [answered]
I've set up several automations, most of them working quite well, but one automation with the trigger "When a new record/row is updated in a table" don't seem to be working. At least not how I expect. If I do a test run everything works like expected, but I also edited several rows in the table directly in the PostgreSQL DB as well as with the BB interface, but I never saw a running automation for this.

Is there a way to debug the automations? With a log or something. I'm still using the cloud hosted solution and plan to transfer everything to a docker container, as soon as everything works fine.

**Comment by melohagan** [comment, marked as answer]
Hi @webserviceXXL 

The automations only run in published apps. So the save row trigger will only fire when you click 'Save' within your published app. 

Altering the data outside of the published UI will not trigger automations.
